lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1609,6 +1609,126 @@ static void optimizeQuantizedMaxSplat(Function *F) {
   }
 }
 
+/// \returns A value representing the given \p constant converted
+/// to the destination type \p dstTy. If the conversion is not
+/// possible, this method returns NodeValue().
+static NodeValue convertConstant(Module &mod, Constant &constant,
+                                 TypeRef dstTy) {
+  // Sort out the easy case first.
+  if (constant.getType() == dstTy) {
+    return NodeValue(&constant, 0);
+  }
+  auto modifyConstantTyAndGet = [&]() -> Constant & {
+    Constant *oneUseCst = getUniquelyUsedConstant(&mod, constant);
+    assert(oneUseCst &&
+           "We should always be able to get a constant from a constant!");
+    oneUseCst->setType(0, dstTy);
```

Comment:
I am not sure that I understand this part of the code. We are setting the type of the constant here. Does this convert the content of the buffer? Will it turn the Int8 backing tensor to Fp32?

Reply:
That's something I discovered when I wrote this code.
Basically we have the type of the constant in two places: the node and the tensor.
This part of the code changes the type of the constant and it doesn't have any effect on the tensor itself.
I should add a comment here and we should probably fix this dual type thing at some point!

Reply:
Comment added.

Reply:
And add something to the verifier.

Reply:
Good point! Done in PR #1869.